File: python/day1.py
from itertools import combinations

input_file = open('../_inputs/day1.txt', 'r')

numbers = [int(line.strip()) for line in input_file.readlines()]

input_file.close()


def run_part1():
    for n1, n2 in combinations(numbers, 2):
        if n1 + n2 == 2020:
            print(n1 * n2)
            break


def run_part2():
    for n1, n2, n3 in combinations(numbers, 3):
        if n1 + n2 + n3 == 2020:
            print(n1 * n2 * n3)
            break

File: rust/src/day1.rs
use itertools::Itertools;

use super::utils;

pub fn run_part1() -> usize {
  let combination = utils::file::read_lines("../_inputs/day1.txt")
    .map(|line| line.parse::<usize>().unwrap())
    .combinations(2)
    .find(|combination| combination.iter().sum::<usize>() == 2020)
    .unwrap();

  let result = combination[0] * combination[1];
  
  println!("{}", result);

  result
}

pub fn run_part2() -> usize {
  let combination = utils::file::read_lines("../_inputs/day1.txt")
    .map(|line| line.parse::<usize>().unwrap())
    .combinations(3)
    .find(|combination| combination.iter().sum::<usize>() == 2020)
    .unwrap();
  
  let result = combination[0] * combination[1] * combination[2];

  println!("{}", result);

  result
}
